docs/ruleset_functions/get_primary_secondary_loops.md

```diff
@@ -0,0 +1,66 @@
+
+Description: Get the list of primary and secondary loops for CHW for a B-RMR.
+
+Inputs:  
+- **B-RMR**: B-RMR that needs to get the list of primary and secondary loops.
+
+Returns: 
+- **primary_secondary_loop_dictionary**: A dictionary that saves pairs of primary and secondary loops for baseline chilled water system, e.g. {primary_loop_1.id: [secondary_loop_1.id, secondary_loop2.id], primary_loop_2.id: [secondary_loop3.id]]}. If B-RMR does not have primary-secondary loop configuration setup, return an empty dictionary.
+
+Logic:  
+
+- Get B-RMR system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMR)`
+
+- For each chiller in B-RMR, save its cooling loop to CHW chiller loop array to get all loops connected to chiller(s): `for chiller in B-RMR.RulesetModelInstance.chillers: chiller_loop_array.append(chiller.cooling_loop)`
+
+- For each building segment in B-RMR: `for building_segment in B-RMR...building_segments:`
+
+  - For each HVAC system in building segment: `for hvac in building_segment.heating_ventilation_air_conditioning_systems`
+
+    - Check if HVAC system is baseline system Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b: `if any(hvac.id in baseline_hvac_system_dict[sys_type] for sys_type in ["SYS-7", "SYS-8", "SYS-11.1", "SYS-11.2", "SYS-12", "SYS-13", "SYS-7B", "SYS-8B", "SYS-11B", "SYS-12B", "SYS-13B"]):`
+
+      - Save chilled water loop that serves cooling systems to non-process CHW coil loop array (array for all loops connected to cooling coils): `chw_coil_loop_array.append(hvac.cooling_system.chilled_water_loop)`
+
+- For each fluid loop in B-RMR: `for fluid_loop in B-RMR.RulesetModelInstance.fluid_loops:`
+
+  - Check if loop type is cooling: `if fluid_loop.type == "COOLING":`
+
+    - Check if loop is connected to both chiller(s) and cooling coil(s), break logic and return an empty dictionary: `if ( fluid_loop.id in chiller_loop_array ) AND ( fluid_loop in chw_coil_loop_array ): BREAK and return primary_secondary_loop_dictionary = {}`
+
+    - Else if loop is connected to chiller(s) only: `else if fluid_loop.id in chiller_loop_array:`
+
+      - Check if all child loops of loop serve baseline system Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b only (to exclude CHW loop served by process chiller(s)): `if child_loop_id in chw_coil_loop_array for child_loop_id in fluid_loop.child_loops:`
+
+        - Save loop to primary loop array: `primary_loop_array.append(fluid_loop.id)`
+
+        - Save all child loops to child loop array: `child_loop_array.append(child_loop_id for child_loop_id in fluid_loop.child_loops)`
+
+    - Else if loop is connected to baseline system cooling coil(s) only: `else if fluid_loop in chw_coil_loop_array:`
+
+      - Check if loop has any child loop, break logic and return an empty dictionary: `if fluid_loop.child_loops != NULL: BREAK and return primary_secondary_loop_dictionary = {}`
```

Comment:
Maybe not terminate the function in this case but just not count this loop as secondary loop in the dictionary returned by the function?